.github/workflows/windows.yml

```diff
@@ -25,7 +25,11 @@ on:
   workflow_dispatch:
 
 jobs:
+  formatting:
+    uses: ./.github/workflows/formatting.yml
   windows:
+    # Shortcircuit and don't burn CI time when formatting will reject
+    needs: formatting
```

Comment:
Is this forcing CI to always run formatting before a build? If so, how? The workflow name is "Formatting".

Reply:
The workflow name is "Formatting" indeed, but that is only the display name of the workflow if it were to run by itself, which it doesn't, because the only `on` flavor is [`on.workflow_call`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_call), which means only other workflows will call it (aka. [reusable worfklow](https://docs.github.com/en/actions/using-workflows/reusing-workflows)). The calling workflow will give it a new name (if it doesn't, it'll be an ugly yml filename). The `needs` part has to refer to to the [`jobs.<job_id>`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_id) of the workflow being called in the _caller's_ yaml file.

For an example of how such a workflow executes, check [this](https://github.com/MathiasMagnus/Test-Packaging/actions/runs/9414364871) run from a toy repo of mine. Hope this clears it up.

Reply:
I think I see. `needs:` refers to the job name `formatting` within the windows.yml file. Correct? The workflow name given in formatting.yml is unused here. Correct?

Reply:
Yes.